### ISSUE TITLE
False positive error when class uses metaclass that derives from `Unknown`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7260]
The error on the last line is a false positive.

```python
from dummy import UnknownBaseClass

class Meta(UnknownBaseClass, type):
    pass

class Settings(metaclass=Meta):
    pass

x: Settings | None # Expected type expression but received "UnionType"
```

**Comment by erictraut** [comment]
This comes from https://github.com/microsoft/pyright/discussions/7258.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.351, which I just published.
